lib/db/DB.js

```diff
@@ -16,19 +19,61 @@ class DB {
   }
 
   define() {
-    this.Order = this.sequelize.define('order', {
-      id: { type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true },
-      peerId: Sequelize.INTEGER,
-      invoice: Sequelize.STRING,
-      quantity: Sequelize.DECIMAL(14, 8),
-      price: Sequelize.DECIMAL(9, 8),
+    // P2P
+
+    this.Peer = this.sequelize.define('peers', {
+      id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
+      nodeKey: { type: DataTypes.STRING, unique: true },
+      ipv4: DataTypes.STRING(15),
+      port: DataTypes.SMALLINT,
     });
 
-    this.Peer = this.sequelize.define('peer', {
-      id: { type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true },
-      nodeKey: { type: Sequelize.STRING, unique: true },
-      ipv4: Sequelize.STRING(15),
-      port: Sequelize.SMALLINT,
+    // DOB
+
+    this.Currency = this.sequelize.define('currencies', {
+      id: { type: DataTypes.STRING, primaryKey: true },
+    });
+
+    //
+
+    this.Pair = this.sequelize.define('pairs', {
+      id: { type: DataTypes.STRING, primaryKey: true },
+      baseCurrency: { type: DataTypes.STRING, allowNull: false },
+      quoteCurrency: { type: DataTypes.STRING, allowNull: false },
+      swapProtocol: {
+        type: DataTypes.ENUM, values: Object.values(enums.swapProtocols), allowNull: true,
+      },
+    });
+
+    this.Pair.belongsTo(this.Currency, {
+      foreignKey: 'baseCurrency',
+    });
+    this.Pair.belongsTo(this.Currency, {
+      foreignKey: 'quoteCurrency',
+    });
+    const derivePairId = (pair) => {
+      pair.id = `${pair.baseCurrency}/${pair.quoteCurrency}`; // eslint-disable-line no-param-reassign
+    };
+    this.Pair.beforeBulkCreate(pairs => pairs.forEach(pair => derivePairId(pair)));
+    this.Pair.beforeCreate(pair => derivePairId(pair));
+
+    //
+
+    this.Order = this.sequelize.define('orders', {
+      id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
+      pairId: { type: DataTypes.STRING, allowNull: false },
+      peerId: { type: DataTypes.INTEGER, allowNull: true },
+      side: {
+        type: DataTypes.ENUM, values: Object.values(enums.orderSides), allowNull: true,
```

Comment:
@kilrau 

sangaman suggested:

> Rather than specify a side, we can do what some other exchanges out there do which is to have a negative quantity indicate a sell while a positive quantity indicates a buy. One less column for us to track.

orderSides: {BUY,SELL}

What do you say about it?

Reply:
Let's do it - positive quantity for buy and negative quantity for sell.